File: python/Impl/ImageFilterProvider.py
#!/usr/bin/env python
# -*- coding:utf-8 -*-
# ImageFilterProvider.py
# 2021, Lin Zhijun, https://github.com/toolgood/ToolGood.TextFilter.Api
# MIT Licensed 
 
__all__ = ['ImageFilterProvider']
import json
import http.client, urllib.parse
import requests


class ImageFilterProvider():

    def __init__(self,textFilterConfig):
        self.__textFilterConfig=textFilterConfig
        self.__ImageLocalFilterUrl = "/api/image-local-filter"
        self.__ImageUploadFilterUrl = "/api/image-upload-filter"
        self.__ImageLocalClassifyUrl = "/api/image-local-classify"
        self.__ImageUploadClassifyUrl = "/api/image-upload-classify"
        self.__BrowserFilterUrl = "/api/image-browser-filter"
        self.__BrowserClassifyUrl = "/api/image-browser-classify"

    def ImageLocalFilter(self,path:str):
        """本地图片检测

        @path      图片文件

        @url       异步回调地址 可空

        @requestId 请求标识，为空时会自动生成
        """
        dictionary = {'path':path}
        result=doPost(self.__textFilterConfig,self.__ImageLocalFilterUrl,dictionary)
        return json.loads(result)

    def ImageUploadFilter(self,path:str):
        """上传图片检测
        
        @path      图片文件

        @url       异步回调地址 可空

        @requestId 请求标识，为空时会自动生成
        """
        u=self.__textFilterConfig.GetTextFilterHost()+self.__ImageUploadFilterUrl

        files = {'file': open(path, 'rb')}
        result = requests.post(u, files=files)
        return json.loads(result.text)

    def ImageLocalClassify(self,path:str):
        """        本地图片分类
        
        @path      图片文件

        @url       异步回调地址 可空

        @requestId 请求标识，为空时会自动生成
        """
        dictionary = {'path':path}
        result=doPost(self.__textFilterConfig,self.__ImageLocalClassifyUrl,dictionary)
        return json.loads(result)

    def ImageUploadClassify(self,path:str):
        """     上传图片分类
     
        @path      图片文件

        @url       异步回调地址 可空

        @requestId 请求标识，为空时会自动生成
        """
        u=self.__textFilterConfig.GetTextFilterHost()+self.__ImageUploadClassifyUrl
        files = {'file': open(path, 'rb')}
        result = requests.post(u, files=files)
        return json.loads(result.text)

    def BrowserFilter(self,vals:list):
        """ 浏览器图片检测
      
        @vals      值

        @url       异步回调地址 可空
        
        @requestId 请求标识，为空时会自动生成
        """
        dictionary = {'params':vals}
        result=doPost(self.__textFilterConfig,self.__BrowserFilterUrl,dictionary)
        return json.loads(result)

    def BrowserClassify(self,vals:list):
        """      浏览器图片分类

        @vals      值

        @url       异步回调地址 可空

        @requestId 请求标识，为空时会自动生成
        """
        dictionary = {'params':vals}
        result=doPost(self.__textFilterConfig,self.__BrowserClassifyUrl,dictionary)
        return json.loads(result)
  

def doPost(textFilterConfig,url,postData):
    u=textFilterConfig.GetTextFilterHost()+url
    result = requests.post(u, postData)
    return  result.text

File: golang/ToolGood/TextFilter/Api/Impl/ImageFilterProvider.go
package Impl


import "encoding/json"
import "net/http"
import "net/url"
import	"io/ioutil"
import	. "../Datas/Images"
import	"strconv"
import	"strings"


type ImageFilterProvider struct{
	textFilterHost string
}

func NewImageFilterProvider(config string) *ImageFilterProvider{
	provider := &ImageFilterProvider{}
	provider.textFilterHost = config
	return  provider;
}

const imageLocalFilterUrl string = "/api/image-local-filter";
const imageUploadFilterUrl string = "/api/image-upload-filter";
const imageLocalClassifyUrl string = "/api/image-local-classify";
const imageUploadClassifyUrl string = "/api/image-upload-classify";
const browserFilterUrl string = "/api/image-browser-filter";
const browserClassifyUrl string = "/api/image-browser-classify";

func (this *ImageFilterProvider)ImageLocalFilter(path string) *ImageFilterResult  {
	u:=this.textFilterHost+imageLocalFilterUrl;

	resp, err := http.PostForm(u, url.Values{"path":{path}}) 
    if err != nil {
		return nil
    }
	defer resp.Body.Close() 
    body,err := ioutil.ReadAll(resp.Body)
    if err != nil {
		return nil
    }
	var result *ImageFilterResult
	err = json.Unmarshal([]byte(body), &result)
	if err != nil {
		return nil
	}
	return result
}
func (this *ImageFilterProvider)ImageUploadFilter(path string) *ImageFilterResult  {
	u:=this.textFilterHost+imageUploadFilterUrl;

	resp, err := http.PostForm(u, url.Values{"path":{path}}) 
    if err != nil {
		return nil
    }
	defer resp.Body.Close() 
    body,err := ioutil.ReadAll(resp.Body)
    if err != nil {
		return nil
    }
	var result *ImageFilterResult
	err = json.Unmarshal([]byte(body), &result)
	if err != nil {
		return nil
	}
	return result
}
func (this *ImageFilterProvider)ImageLocalClassify(path string) *ImageClassifyResult  {
	u:=this.textFilterHost+imageLocalClassifyUrl;

	resp, err := http.PostForm(u, url.Values{"path":{path}}) 
    if err != nil {
		return nil
    }
	defer resp.Body.Close() 
    body,err := ioutil.ReadAll(resp.Body)
    if err != nil {
		return nil
    }
	var result *ImageClassifyResult
	err = json.Unmarshal([]byte(body), &result)
	if err != nil {
		return nil
	}
	return result
}
func (this *ImageFilterProvider)ImageUploadClassify(path string) *ImageClassifyResult  {
	u:=this.textFilterHost+imageUploadClassifyUrl;

	resp, err := http.PostForm(u, url.Values{"path":{path}}) 
    if err != nil {
		return nil
    }
	defer resp.Body.Close() 
    body,err := ioutil.ReadAll(resp.Body)
    if err != nil {
		return nil
    }
	var result *ImageClassifyResult
	err = json.Unmarshal([]byte(body), &result)
	if err != nil {
		return nil
	}
	return result
}
func (this *ImageFilterProvider)BrowserFilter(vals []float32) *ImageFilterResult  {
	u:=this.textFilterHost+browserFilterUrl;

	resp, err := http.PostForm(u, url.Values{"Params":{toStr(vals)}}) 
    if err != nil {
		return nil
    }
	defer resp.Body.Close() 
    body,err := ioutil.ReadAll(resp.Body)
    if err != nil {
		return nil
    }
	var result *ImageFilterResult
	err = json.Unmarshal([]byte(body), &result)
	if err != nil {
		return nil
	}
	return result
}
func (this *ImageFilterProvider)BrowserClassify(vals []float32) *ImageClassifyResult  {
	u:=this.textFilterHost+browserClassifyUrl;

	resp, err := http.PostForm(u, url.Values{"Params":{toStr(vals)}}) 
    if err != nil {
		return nil
    }
	defer resp.Body.Close() 
    body,err := ioutil.ReadAll(resp.Body)
    if err != nil {
		return nil
    }
	var result *ImageClassifyResult
	err = json.Unmarshal([]byte(body), &result)
	if err != nil {
		return nil
	}
	return result
}

func toStr(vals []float32) string{
	var builder strings.Builder

	for i := 0; i < len(vals) ; i++ {
		val:=vals[i]
		builder.WriteString(strconv.FormatFloat(float64(val), 'f', 8, 64))
	}
	return  builder.String()
}


